src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -64,41 +98,98 @@ protected void buildLanguageFoldRegions(@NotNull List<FoldingDescriptor> descrip
         } catch (CancellationException e) {
             // cancel the LSP requests textDocument/foldingRanges
             foldingRangeSupport.cancel();
-            return;
+            return Collections.emptyList();
         } catch (ExecutionException e) {
             LOGGER.error("Error while consuming LSP 'textDocument/foldingRanges' request", e);
-            return;
+            return Collections.emptyList();
         }
 
-        if (isDoneNormally(foldingRangesFuture)) {
-            // textDocument/foldingRanges has been collected correctly, create list of IJ FoldingDescriptor from LSP FoldingRange list
-            List<FoldingRange> foldingRanges = foldingRangesFuture.getNow(null);
-            if (foldingRanges != null) {
-                for (FoldingRange foldingRange : foldingRanges) {
-                    int start = getStartOffset(foldingRange, document);
-                    int end = getEndOffset(foldingRange, document);
-                    if (start >= end) {
-                        // same line, ignore the folding
-                        continue;
-                    }
-                    String collapsedText = foldingRange.getCollapsedText();
-                    if (collapsedText != null) {
-                        descriptors.add(new FoldingDescriptor(root.getNode(), new TextRange(start, end), null, collapsedText));
-                    } else {
-                        descriptors.add(new FoldingDescriptor(root.getNode(), new TextRange(start, end)));
-                    }
+        if (!isDoneNormally(foldingRangesFuture)) {
+            return Collections.emptyList();
+        }
+
+        // textDocument/foldingRanges has been collected correctly, create list of IJ FoldingDescriptor from LSP FoldingRange list
+        return foldingRangesFuture.getNow(null);
+    }
+
+    @Nullable
+    private static TextRange getTextRange(@NotNull FoldingRange foldingRange, @NotNull Document document) {
+        return getTextRange(foldingRange, document, null, null);
+    }
+
+    /**
+     * Returns the IDE text range for the LSP folding range, optionally bounded by the expected paired open/close braces.
+     *
+     * @param foldingRange   the LSP folding range
+     * @param document       the document
+     * @param openBraceChar  the optional open brace character
+     * @param closeBraceChar the optional paired close brace character
+     * @return the corresponding IDE text range, or null if no valid text range could be derived
+     */
+    @Nullable
+    static TextRange getTextRange(
+            @NotNull FoldingRange foldingRange,
+            @NotNull Document document,
+            @Nullable Character openBraceChar,
+            @Nullable Character closeBraceChar
+    ) {
+        TextRange textRange = null;
+
+        String documentText = document.getText();
+        int documentLength = documentText.length();
+
+        int start = LSPFoldingRangeBuilder.getStartOffset(foldingRange, document);

```

Comment:
`getStartOffset()` and `getEndOffset()` are already going to be adjusted as they were previously if the start/end characters aren't specified by the language server, so that behavior is retained.